module/input/Camera/src/Camera.cpp

```diff
@@ -54,7 +54,8 @@ namespace module::input {
              ************************************************************************/
             if (it == cameras.end()) {
                 // Strip the .yaml off the name of the file to get the name of the camera
-                std::string name = ::basename(config.fileName.substr(0, config.fileName.find_last_of('.')).c_str());
+                const std::string config_file = config.fileName.string();
+                const std::string name = ::basename(config_file.substr(0, config_file.find_last_of('.')).c_str());
```

Comment:
Why? Perhaps [this](https://en.cppreference.com/w/cpp/filesystem/path/stem) or similar would be better?